File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/VfsRoot.java
/**
 * @file
 * @brief Vfs root directory interface
 * @author vitamin.caig@gmail.com
 */

package app.zxtune.fs;

import android.net.Uri;
import androidx.annotation.Nullable;

import java.io.IOException;

interface VfsRoot extends VfsDir {

  /**
   * Tries to resolve object by uri
   *
   * @param uri Identifier to resolve
   * @return Found object or null otherwise
   */
  @Nullable
  VfsObject resolve(Uri uri) throws IOException;
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/VfsRoot.kt
/**
 * @file
 * @brief Vfs root directory interface
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.fs

import android.net.Uri
import java.io.IOException

internal interface VfsRoot : VfsDir {
    /**
     * Tries to resolve object by uri
     *
     * @param uri Identifier to resolve
     * @return Found object or null otherwise
     */
    @Throws(IOException::class)
    fun resolve(uri: Uri): VfsObject?
}
